src/coprocessor/dag/executor/table_scan.rs

```diff
@@ -68,7 +68,12 @@ impl<'a> TableScanExecutor<'a> {
             None => return Ok(None),
         };
         let h = box_try!(table::decode_handle(&key));
-        let row_data = box_try!(table::cut_row(value, &self.col_ids));
+        let mut row_data = box_try!(table::cut_row(value, &self.col_ids));
+        if self.col_ids.contains(&EXTRA_HANDLE_COLUMN_ID) {
+            let handle_datum = datum::Datum::I64(h);
+            let mut bytes = box_try!(datum::encode_key(&[handle_datum]));
```

Comment:
Could we pick `encode_handle` directly from `key`?   And could we remove `handle` from struct `Row`?